### ISSUE TITLE
Mesh dense changes significantly

### ISSUE THREAD

**Issue by aribal2018** [CHLNDDEV/OceanMesh2D#304]
Dear All,

I am working in a bay and creating the mesh using OceanMesh2D. At the open ocean, I set the maximum resolution to be 2.5 km and the coastline is 100 meters.

As seen from my attached figure, the meshes are almost homogenous from the left until the middle of the bay (circled red). Then, the resolutions of the mesh suddenly change along the coastline which is probably one km from the coastline. In my opinion, this sudden change is probably not good.

![mesh](https://github.com/CHLNDDEV/OceanMesh2D/assets/39908089/92687d04-b77a-45aa-8b0d-ba94b1f319c8)

I am actually expecting that the dense of the mesh start increasing when entering the bay. Say, for example, I will expect that the dense of the mesh will start to increase from the longitude 237.5  until the coastline. By doing this, there will be a "smooth" change in the mesh resolutions.

Therefore, I was wondering if there an option to do that.
Hope this makes sense.
Thank you very much.

**Comment by krober10nd** [comment]
Adjust the gradation rate to a lower value, such as 0.05 or 0.075, and/or constrain the minimum mesh element size according to depth. For syntax guidance on depth-based bounding, refer to the Galveston Bay example.

**Comment by aribal2018** [comment]
Hi Keith,

Happy New Year.

I looked at the example but I still could not get the mesh that I am expecting.
I tried:
fh = edgefx('geodata',gdat,'dis',0.3,'fs',3,'max_el_ns',200,'max_el',100,'g',0.1);

This actually violate the value of the grade which is supposed to be between 0.2 and 0.3. The mesh is slightly improved but the transition is still very high.

![image](https://github.com/CHLNDDEV/OceanMesh2D/assets/39908089/fb7bd541-e1ab-439c-aad1-b88475bb211f)

About nine years ago, I created a mesh using SMS and the transition is very good. I hope I can use OceanMesh2D to get a similar results.

![mesh](https://github.com/CHLNDDEV/OceanMesh2D/assets/39908089/00255089-2cc6-4790-9b41-125f2a50211a)

As can be seen from this mesh, the mesh transition location change is quite far from the island. Hence, the transition from coarser mesh to the finer mesh is quite smooth.

I do hope, I can get a similar results using OceanMesh2D as this method is quite convenient to use.
